### PR TITLE
Add Matterbridge LXC

### PR DESCRIPTION
**Pull request by MickLesk**
## Description
New Matterbridge LXC added

- Tested and Used in my Homelab
- Update is available in the script, but it also works very well via the web-gui. (recommended)

Port: 8283
Source: https://github.com/Luligu/matterbridge/

A short code review would be nice. So far everything works, I have removed unnecessary packages and updated NodeJS to the latest dearmor gpg version. 

Request: https://github.com/tteck/Proxmox/discussions/50#discussioncomment-9743382

image

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] New Script (Develop a new script or set of scripts that are fully functional and thoroughly tested)
- [x] I have performed a self-review of my code, adhering to established codebase patterns and conventions.
- [x] This change requires a documentation update

### CODE REVIEW

File: ct/matterbridge.sh [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   /  |/  /___ _/ /_/ /____  _____/ /_  _____(_)___/ /___ ____ 
+  / /|_/ / __ `/ __/ __/ _ \/ ___/ __ \/ ___/ / __  / __ `/ _ \
+ / /  / / /_/ / /_/ /_/  __/ /  / /_/ / /  / / /_/ / /_/ /  __/
+/_/  /_/\__,_/\__/\__/\___/_/  /_.___/_/  /_/\__,_/\__, /\___/ 
+                                                  /____/                                      
+EOF
+}
+header_info
+echo -e "Loading..."
+APP="matterbridge"
```
Comment (tteck) [inline ct/matterbridge.sh]: Matterbridge

File: ct/matterbridge.sh [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  SD=""
+  NS=""
+  MAC=""
+  VLAN=""
+  SSH="no"
+  VERB="no"
+  echo_default
+}
+
+function update_script() {
```
Comment (tteck) [inline ct/matterbridge.sh:55]:
```
function update_script() {
header_info
if [[ ! -d /opt/matterbridge ]]; then msg_error "No ${APP} Installation Found!"; exit; fi

RELEASE=$(curl -s https://api.github.com/repos/Luligu/matterbridge/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3)}')
if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
  msg_info "Stopping ${APP} Service"
  systemctl stop matterbridge
  msg_ok "Stopped ${APP} Service"

  msg_info "Updating ${APP} to ${RELEASE}"
  cd /opt/matterbridge
  wget -q "https://github.com/Luligu/matterbridge/archive/refs/tags/${RELEASE}.zip" 
  unzip -q ${RELEASE}.zip
  mv matterbridge-${RELEASE} /opt/matterbridge
  cd /opt/matterbridge
  npm ci >/dev/null 2>&1
  npm run build >/dev/null 2>&1
  echo "${RELEASE}" >/opt/${APP}_version.txt
  msg_ok "Updated ${APP} to ${RELEASE}"

  msg_info "Cleaning up"
  rm /opt/${RELEASE}.zip 
  msg_ok "Cleaned"

  msg_info "Starting ${APP} Service"
  systemctl start matterbridge
  sleep 1
  msg_ok "Started ${APP} Service"
  msg_ok "Updated Successfully!\n"
else
  msg_ok "No update required. ${APP} is already at ${RELEASE}"
fi
exit
}
```

File: install/matterbridge-install.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+color
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies (Patience)"
+$STD apt-get install -y --no-install-recommends \
+  unzip \
```
Comment (tteck) [inline install/matterbridge-install.sh]: Not needed. `unzip` is included in `mc`

File: install/matterbridge-install.sh [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+msg_info "Install Matterbridge" 
+RELEASE=$(curl -s https://api.github.com/repos/Luligu/matterbridge/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3) }')
+wget -q "https://github.com/Luligu/matterbridge/archive/refs/tags/${RELEASE}.zip"
+unzip -q ${RELEASE}.zip
+mv matterbridge-${RELEASE} /opt/matterbridge
+rm -R ${RELEASE}.zip 
+cd /opt/matterbridge
+$STD npm ci
+$STD npm run build
```
Comment (tteck) [inline install/matterbridge-install.sh:50]: add `echo "${RELEASE}" >/opt/${APPLICATION}_version.txt` after `$STD npm run build`

File: install/matterbridge-install.sh [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+systemctl enable -q --now matterbridge.service
+msg_ok "Created Service"
+
+motd_ssh
+customize
+
+msg_info "Cleaning up"
+$STD apt-get autoremove
+$STD apt-get autoclean
+msg_ok "Cleaned"
```
Comment (tteck) [inline install/matterbridge-install.sh:85]: `$STD apt-get -y autoremove`
`$STD apt-get -y autoclean`

**Comment by MickLesk** [comment]
Nice, thx for Feedback. I Check after the lunch break :-)

**Pull request closed without merging**